servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultBlockingStreamingHttpServerResponse.java

```diff
@@ -132,8 +135,19 @@ public BlockingStreamingHttpServerResponse context(final ContextMap context) {
         return this;
     }
 
+    @Nonnull
+    @Override
+    public ContextMap context() {
+        final ContextMap ctx = super.context();
```

Comment:
@idelpivnitskiy of curse this will not cover the case where someone grabs the context, then sends, and then writes into it. So we likely need a "SendAwareContextMap" and always return that one wrapped.

Reply:
Yes, I think whenever this method is invoked it should hold a reference to `this` response and always run `response.checkSent()` for any modifications. Or it can just take a lambda that will be referenced by `this::checkSent()`.